aten/src/ATen/FunctionalTensorWrapper.cpp

```diff
@@ -533,6 +534,16 @@ void commit_update(ITensorListRef functional_tensor) {
   }
 }
 
+void mark_mutated_by_triton_kernel(const Tensor& functional_tensor) {
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(isFunctionalTensor(functional_tensor));
```

Comment:
These can just be `TORCH_CHECK` instead of `TORCH_INTERNAL_ASSERT_DEBUG_ONLY`:

(1) Since these API's can be used potentially by someone outside of pytorch (they're bound to python), it's more of a user error than an internal assert

(2) they probably shouldn't be debug only (the _DEBUG_ONLY means that these asserts will only run in debug builds of pytorch, so they won't fire at all in our normal builds. These are meant for perf-critical code paths, but this code doesn't need to be too fast).